csaf-lib/src/csaf/csaf2_0/getter_implementations.rs

```diff
@@ -0,0 +1,78 @@
+use std::collections::BTreeSet;
+use std::ops::Deref;
+use std::str::FromStr;
+use crate::csaf::csaf2_0::schema::{CommonSecurityAdvisoryFramework, ProductGroup, ProductTree, Remediation, Vulnerability};
+use crate::csaf::csaf2_1::schema::CategoryOfTheRemediation as Remediation21;
+use crate::csaf::getter_traits::{CsafTrait, ProductGroupTrait, ProductTreeTrait, RemediationTrait, VulnerabilityTrait};
+use crate::csaf::helpers::resolve_product_groups;
+
+impl RemediationTrait for Remediation {
```

Comment:
Should we move this file to a more "neutral" place? It is a little bit confusing that we now import csaf 2.1 schema things into a csaf 2.0 folder.

Also needs a documentation

Reply:
> This is a normalization step. Since 2.1 extends the set but keeps identifiers of 2.0, we can do it like this. Otherwise, we would need to directly use `String`s instead of `enum`s or implement our own `enum`, which is both non-ideal...

But that would mean that we accidentally would parse something as "CSAF 2.1" in CSAF 2.0, e.g., if 2.1 has an identifier which 2.0 does not have. Probably it would be better to use 2.0 structs in 2.1 and "extend" them somehow, but not the other way around